### PR TITLE
feat: add voucher categories list

### PR DESCRIPTION
**Pull request by ardasnturk**
- added `GET_VOUCHERS_CATEGORIES` query to list only the category information of the voucher
- added `parseVouchersCategories` parser to `listItemParser` so that we can show the category list of the voucher in the format we want
- changed query in `parseListItemsFromQuery` to parse voucher category in `VoucherIndexScreen`
- added control in `ListHeaderComponent` to show only query value in `Vouchers`
- added count to `TextListItem` to show how many vouchers are in which category of the voucher on the list

SVAK-35

## Screenshots:

|VoucherIndexScreen `query === vouchers`|VoucherIndexScreen `query === vouchersCategories`|
|--|--|
Simulator Screenshot - iPhone 14 Pro - 2024-01-09 at 14 47 40|Simulator Screenshot - iPhone 14 Pro - 2024-01-09 at 14 47 43

### CODE REVIEW

File: src/components/TextListItem.tsx [hunk 1]
```diff
@@ -146,7 +148,10 @@ export const TextListItem: NamedExoticComponent<Props> & {
         ) : undefined)}
 
       {!!navigation && !withCard && (
-        <Icon.ArrowRight color={colors.darkText} size={normalize(18)} />
+        <>
+          {!!count && <BoldText>({count})</BoldText>}
```
Comment (donni106) [inline src/components/TextListItem.tsx:152]: if we place the count on the right, it should not be surrounded by brackets

Comment (ardasnturk) [reply src/components/TextListItem.tsx:152]: Yes, you are right. I'll fix it. 👍

**Review by donni106 — APPROVED** [review]

File: src/components/TextListItem.tsx (hunk 1 above)
Comment (donni106) [reply src/components/TextListItem.tsx]: no, i meant we can print 1 instead of (1)

Comment (ardasnturk) [reply src/components/TextListItem.tsx]: Ach sorry I misunderstood. I used `()` to make the application look the same as the category list. But you are right. If we show it on the right side, we don't need `()`.

**Pull request merged**